apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go

```diff
@@ -176,12 +177,19 @@ func (api *CrossModelRelationsAPI) registerRemoteRelation(relation params.Regist
 		return nil, errors.Annotate(err, "loading model")
 	}
 	offerURL := crossmodel.MakeURL(model.Owner().Name(), model.Name(), relation.OfferName, "")
-	if err := api.checkMacaroons(relation.Macaroons, map[string]string{
+	attr, err := api.checkMacaroons(relation.Macaroons, map[string]string{
 		"source-model-uuid": api.st.ModelUUID(),
 		"offer-url":         offerURL,
-	}); err != nil {
+	})
+	if err != nil {
 		return nil, err
 	}
+	// The macaroon needs to be attenuated to a user.
+	username, ok := attr["username"]
```

Comment:
All we are doing here is extracting the username which was attached to the macaroon when it was minted. This is used to populate the offer connection details so we know who created this relation. There will be a separate discharge done (eventually) to check that the username has permission to consume the offer. That check was done when the consume was first executed, but may have changed between then and now when the relation is set up.